packages/loopring_v3/contracts/impl/libtransactions/TransferTransaction.sol

```diff
@@ -0,0 +1,129 @@
+// SPDX-License-Identifier: Apache-2.0
+// Copyright 2017 Loopring Technology Limited.
+pragma solidity ^0.6.10;
+pragma experimental ABIEncoderV2;
+
+import "../../iface/ExchangeData.sol";
+import "../../thirdparty/BytesUtil.sol";
+import "../../lib/EIP712.sol";
+import "../../lib/FloatUtil.sol";
+import "../../lib/MathUint.sol";
+import "../../lib/SignatureUtil.sol";
+
+
+/// @title TransferTransaction
+/// @author Brecht Devos - <brecht@loopring.org>
+library TransferTransaction
+{
+    using BytesUtil            for bytes;
+    using FloatUtil            for uint;
+    using MathUint             for uint;
+    using SignatureUtil        for bytes32;
+
+    bytes32 constant public TRANSFER_TYPEHASH = keccak256(
+        "Transfer(address from,address to,uint16 tokenID,uint256 amount,uint16 feeTokenID,uint256 fee,uint256 data,uint32 nonce)"
+    );
+
+    /*event ConditionalTransferConsumed(
+        address indexed from,
+        address indexed to,
+        uint16          token,
+        uint            amount
+    );*/
+
+    function process(
```

Comment:
Brecht, is receipt supported in this version? https://github.com/Loopring/protocols/issues/1279

Reply:
Yes, `IExchangeV3.getBlockInfo` can be used onchain to get the public data hash when `storeDataHashOnchain` is used by the operator. So we can build logic on top of it in other smart contracts that implement the actual receipts.